src/hooks/core/auth/useUserAuthHook.ts

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import {useRecoilState} from 'recoil';
+
+import profileAtom from '../../../atom/profileAtom';
+import {Context} from '../../../context';
+import {ProfileContext} from '../../../../types/context/profilecontext.types';
+
+const useUserAuthHook = () => {
+  const [profile] = React.useContext(Context).profile;
+  const profileContext = profile as ProfileContext;
+
+  const [authAtomState, setAuth] = useRecoilState(profileAtom);
+
+  const {anonProfileId, signedProfileId, token, anonymousToken} = authAtomState;
+
+  const resetAuth = () =>
+    setAuth({
+      signedProfileId: null,
+      anonProfileId: null,
+      token: null,
+      anonymousToken: null
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `resetAuth` function resets all fields in the `authAtomState` to null. This might not be the desired behavior if there are other fields in the state that are not included in this function. Consider resetting only the fields related to authentication.

```diff
-    setAuth({
-      signedProfileId: null,
-      anonProfileId: null,
-      token: null,
-      anonymousToken: null
-    });
+    setAuth(prevState => ({
+      ...prevState,
+      signedProfileId: null,
+      anonProfileId: null,
+      token: null,
+      anonymousToken: null
+    }));
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
no need, should delete all auth atom state if reset